### DISCUSSION TITLE
Not able to retrieve custom field data from SPO list

### DISCUSSION THREAD

**Discussion by AllanMacAshurst** [microsoft-search/pnp-modern-search#4074] [Q&A]
We have installed PNP Modern Search Web Parts in an attempt to build custom list search / filter capability and configured it as per guidance, see below:
![image](https://github.com/user-attachments/assets/4050756c-f6bc-49b4-90e1-93d366af2d6c)

However when we view in debug mode the search output, it returns null for the various fields we want to filter against

**Version used**
Ex: 4.8

**Describe the bug**
When setting Selected Properties in search results web part, it returns null for most fields and for URL field returns wrong value 

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Install PNP Mopdern Search web parts to tenant and then site
2. Add Search Input and Search Results to page
3. Set Query template uto target list items
4. Add Fields to be returned in list view
5. Get nulls in debug view

**Expected behavior**
We expected that the values for each item with include the list item values for configured fields

**Screenshots**
![image](https://github.com/user-attachments/assets/4050756c-f6bc-49b4-90e1-93d366af2d6c)

**Desktop (please complete the following information):**
 - Browser [MS Edge]

**Additional context**

**Comment by kasperbolarsen** [comment]
Can you check if you get any results using one of these debugging tools, https://microsoft-search.github.io/pnp-modern-search/using-query-tools-to-verify-issues/

**Comment by AllanMacAshurst** [comment]
I'll give them a try, but we are able to get data back from list using pnp js in spfx



Allan MacRitchie (He/Him/His)
Senior SharePoint Microsoft 365 Developer
_______________________________________________________

Central Exchange, 20 Waterloo Street, Glasgow, G2 6DB
D: +44 141 375 4741
www.ashurst.com<http://www.ashurst.com>
[Ashurst]  is outpacing change with clients. Find out how<https://www.ashurst.com/en/outpacing-change?utm_source=esig_text&utm_medium=email&utm_campaign=BrandLaunchExt&utm_content=signature_text_1>.

Visit our Powering Change<https://www.ashurst.com/en/powering-change-technologies-fuelling-the-future-report?utm_source=edm&utm_medium=email&utm_campaign=PoweringChange24&utm_content=Email_Signature> webpage for insights on the technologies fuelling the future.

**Comment by kasperbolarsen** [comment]
That is far from the same 😉

Those fields,  are they site columns or list columns?

**Comment by AllanMacAshurst** [comment]
They are list columns



Allan MacRitchie (He/Him/His)
Senior SharePoint Microsoft 365 Developer
_______________________________________________________

Central Exchange, 20 Waterloo Street, Glasgow, G2 6DB
D: +44 141 375 4741
www.ashurst.com<http://www.ashurst.com>
[Ashurst]  is outpacing change with clients. Find out how<https://www.ashurst.com/en/outpacing-change?utm_source=esig_text&utm_medium=email&utm_campaign=BrandLaunchExt&utm_content=signature_text_1>.

Visit our Powering Change<https://www.ashurst.com/en/powering-change-technologies-fuelling-the-future-report?utm_source=edm&utm_medium=email&utm_campaign=PoweringChange24&utm_content=Email_Signature> webpage for insights on the technologies fuelling the future.

**Comment by AllanMacAshurst** [comment]
I appreciate they are far from the same, 
Unfortunately I am unable to use the search tool, I will try the Chrome addin shortly

**Comment by kasperbolarsen** [comment]
In general we recommend using Site columns as the search index will create managed properties for site columns,  unlike list columns where you will have to connect the crawled properties to a managed property your self

**Comment by codingsisya** [comment]
As @kasperbolarsen said, use Site Columns and map to RefinableString if you intend to use those columns as Refinement filters

**Comment by kasperbolarsen** [comment]
In general I recommend reading https://learn.microsoft.com/en-us/microsoft-365/community/how-do-site-columns-become-managed-properties-thus-available-for-search

Carefully,  as requirements like thare have to be a value in the column in order for it to picked up by the index. Likewise,  the requirement that you have to be a Site collection admin on a Team site is also tripping a lot if people
